docs/internals/data_file.md

```diff
@@ -0,0 +1,196 @@
+---
+sidebar_position: 2
+---
+
+# Data File
+
+> “Just show me the tables already!”
+> — probably not Fred Brooks
+
+Each TigerBeetle replica stores all data inside a single file, called the data file (conventional
+extension is `.tigerbeetle`). This document describes the high level layout of the data file. The
+presentation is simplified a bit, to provide intuition without drowning the reader in details.
+Consult the source code for byte-level details!
+
+The data file is divided into several zones, with the main ones being:
+
+- write-ahead log
+- superblock
+- grid
+
+The grid forms the bulk of the data file (up to several terabytes). It is an array of 64KiB blocks:
+
+```zig
+pub const Block = [constants.block_size]u8;
+pub const BlockPtr = *align(constants.sector_size) Block;
+```
+
+The grid serves as a raw storage layer. Higher level data structures (notably, the LSM tree) are
+mapped to physical grid blocks. Because TigerBeetle is deterministic, the contents of the grid is
```

Comment:
```suggestion
mapped to physical grid blocks. Because TigerBeetle is deterministic, the used portion of referenced blocks in the grid is
```

Reply:
I was debating about the level of precision here, as explaining _exactly_ whats going on is tricky, especially given that we haven't introduce LSM or superblock yet. I think "contents" is sufficiently coarse -- gird is elastic, and it logically contains only what's in the superblock.

Reply:
changed to "the used portion of" --- its enough to intuit that the grid it not literally bitwise identical, but doesn't try to explain stuff we are not ready for at this point.